tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -0,0 +1,365 @@
+        Taxonomy.objects.create(name="Taxonomy enabled 2", enabled=True).save()
+        Taxonomy.objects.create(name="Taxonomy disabled", enabled=False).save()
+
+        url = reverse("oel_tagging:taxonomy-list")
+
+        self.client.force_authenticate(user=self.staff)
+        if enabled is not None:
+            response = self.client.get(url, {"enabled": enabled})
+        else:
+            response = self.client.get(url)
+        assert response.status_code == expected_status
+
+        # If we were able to list the taxonomies, check that we got the expected number back
+        if status.is_success(expected_status):
+            assert len(response.data) == expected_count
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_200_OK),
+        ("staff", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_list_taxonomy(self, user_attr, expected_status):
+        url = reverse("oel_tagging:taxonomy-list")
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.get(url)
+        assert response.status_code == expected_status
+
+    @ddt.data(
+        (None, {"enabled": True}, status.HTTP_403_FORBIDDEN),
+        (None, {"enabled": False}, status.HTTP_403_FORBIDDEN),
+        (
+            "user",
+            {"enabled": True},
+            status.HTTP_200_OK,
+        ),
+        ("user", {"enabled": False}, status.HTTP_404_NOT_FOUND),
+        ("staff", {"enabled": True}, status.HTTP_200_OK),
+        ("staff", {"enabled": False}, status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_detail_taxonomy(self, user_attr, taxonomy_data, expected_status):
+        create_data = {**{"name": "taxonomy detail test"}, **taxonomy_data}
+        taxonomy = Taxonomy.objects.create(**create_data)
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": taxonomy.pk})
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.get(url)
+        assert response.status_code == expected_status
+
+        if status.is_success(expected_status):
+            check_taxonomy(response.data, taxonomy.pk, **create_data)
+            self.assertGreaterEqual(response.data.items(), create_data.items())
+
+    def test_detail_taxonomy_404(self):
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": 123123})
+
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.get(url)
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_403_FORBIDDEN),
+        ("staff", status.HTTP_201_CREATED),
+    )
+    @ddt.unpack
+    def test_create_taxonomy(self, user_attr, expected_status):
+        url = reverse("oel_tagging:taxonomy-list")
```

Comment:
Nit: In my opionion, it's better to hard-code the REST API URLs in tests, so that if you change the URL (breaking backwards compatibility), the test will fail. If you use reverse, you can change the URLs which breaks backwards compatibility, but the tests will still pass.

This is an optional piece of feedback.

Reply:
That's a good point.. @rpenido do you have time to make this change to your tests this sprint? If not, then we can leave this out.

I recommend putting the hard-coded strings as constants up at the top and use string formatting, e.g.

TAXONOMY_LIST_URL = `/tagging/rest_api/v1/taxonomy/'
TAXONOMY_VIEW_URL = f`/tagging/rest_api/v1/taxonomy/{taxonomy.pk}'